commit 5be2b1ed55fd7259dfdf2c82e7701dba407b8b62
Author: Thibaud Colas <thibaudcolas@gmail.com>
Date:   Mon Mar 27 12:48:02 2023 +0100

    Change ChooseParentView to use plain text breadcrumbs for page titles, avoiding stored XSS vector

diff --git a/wagtail/contrib/modeladmin/forms.py b/wagtail/contrib/modeladmin/forms.py
index 92feab81f0..be5431a56f 100644
--- a/wagtail/contrib/modeladmin/forms.py
+++ b/wagtail/contrib/modeladmin/forms.py
@@ -1,5 +1,4 @@
 from django import forms
-from django.utils.safestring import mark_safe
 from django.utils.translation import gettext as _
 
 from wagtail.models import Page
@@ -12,7 +11,7 @@ class PageChoiceField(forms.ModelChoiceField):
             obj.get_ancestors(inclusive=True).exclude(depth=1).specific(defer=True)
         ):
             bits.append(ancestor.get_admin_display_title())
-        return mark_safe('<span class="icon icon-arrow-right"></span>'.join(bits))
+        return " | ".join(bits)
 
 
 class ParentChooserForm(forms.Form):
diff --git a/wagtail/contrib/modeladmin/tests/test_page_modeladmin.py b/wagtail/contrib/modeladmin/tests/test_page_modeladmin.py
index 9f9b52d2bc..9b3825c21c 100644
--- a/wagtail/contrib/modeladmin/tests/test_page_modeladmin.py
+++ b/wagtail/contrib/modeladmin/tests/test_page_modeladmin.py
@@ -286,6 +286,21 @@ class TestChooseParentView(TestCase, WagtailTestUtils):
         """
         self.assertContains(response, expected, html=True)
 
+    def test_page_title_html_escaping(self):
+        homepage = Page.objects.get(url_path="/home/")
+        business_index = BusinessIndex(
+            title="Title with <script>alert('XSS')</script>",
+        )
+        homepage.add_child(instance=business_index)
+
+        response = self.client.get("/admin/tests/businesschild/choose_parent/")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertNotContains(response, "Title with <script>alert('XSS')</script>")
+        self.assertContains(
+            response, "Title with &lt;script&gt;alert(&#x27;XSS&#x27;)&lt;/script&gt;"
+        )
+
 
 class TestChooseParentViewForNonSuperuser(TestCase, WagtailTestUtils):
     fixtures = ["test_specific.json"]
